carma_cooperative_perception/src/multiple_object_tracker_component.cpp

```diff
@@ -0,0 +1,483 @@
+{
+  carma_cooperative_perception_interfaces::msg::Track msg;
+
+  msg.header.stamp.sec = mot::remove_units(units::math::floor(track.timestamp));
+  msg.header.stamp.nanosec = mot::remove_units(
+    units::time::nanosecond_t{units::math::fmod(track.timestamp, units::time::second_t{10.0})});
+  msg.header.frame_id = "map";
+
+  msg.id = track.uuid.value();
+  msg.motion_model = msg.MOTION_MODEL_CTRV;
+  msg.pose.pose.position.x = mot::remove_units(track.state.position_x);
+  msg.pose.pose.position.y = mot::remove_units(track.state.position_y);
+
+  tf2::Quaternion orientation;
+  orientation.setRPY(0, 0, mot::remove_units(track.state.yaw.get_angle()));
+  msg.pose.pose.orientation.x = orientation.getX();
+  msg.pose.pose.orientation.y = orientation.getY();
+  msg.pose.pose.orientation.z = orientation.getZ();
+  msg.pose.pose.orientation.w = orientation.getW();
+
+  msg.twist.twist.linear.x = mot::remove_units(track.state.velocity);
+  msg.twist.twist.angular.z = mot::remove_units(track.state.yaw_rate);
+
+  return msg;
+}
+
+static auto to_ros_msg(const Track & track)
+{
+  static constexpr mot::Visitor visitor{
+    [](const mot::CtrvTrack & t) { return to_ros_msg(t); },
+    [](const mot::CtraTrack & t) { return to_ros_msg(t); },
+    [](const auto &) { throw std::runtime_error{"cannot make ROS 2 message from track type"}; }};
+
+  return std::visit(visitor, track);
+}
+
+MultipleObjectTrackerNode::MultipleObjectTrackerNode(const rclcpp::NodeOptions & options)
+: CarmaLifecycleNode{options}
+{
+}
+
+auto MultipleObjectTrackerNode::handle_on_configure(
+  const rclcpp_lifecycle::State & /* previous_state */) -> carma_ros2_utils::CallbackReturn
+{
+  RCLCPP_INFO(get_logger(), "Lifecycle transition: configuring");
+
+  track_list_pub_ = create_publisher<carma_cooperative_perception_interfaces::msg::TrackList>(
+    "output/track_list", 1);
+
+  detection_list_sub_ = create_subscription<
+    carma_cooperative_perception_interfaces::msg::DetectionList>(
+    "input/detection_list", 1,
+    [this](const carma_cooperative_perception_interfaces::msg::DetectionList::SharedPtr msg_ptr) {
+      if (const auto current_state{this->get_current_state().label()}; current_state == "active") {
+        this->store_new_detections(*msg_ptr);
+      } else {
+        RCLCPP_WARN(
+          this->get_logger(),
+          "Trying to receive message on the topic '%s', but the containing node is not activated. "
+          "Current node state: '%s'",
+          this->detection_list_sub_->get_topic_name(), current_state.c_str());
+      }
+    });
+
+  declare_parameter(
+    "execution_frequency_hz", mot::remove_units(units::frequency::hertz_t{1 / execution_period_}));
+
+  declare_parameter(
+    "track_promotion_threshold", static_cast<int>(track_manager_.get_promotion_threshold().value));
+
+  declare_parameter(
+    "track_removal_threshold", static_cast<int>(track_manager_.get_promotion_threshold().value));
```

Comment:
Do we need to have get_parameter() for these to actually extract the values from the parameter?